kpi/deployment_backends/kobocat_backend.py

```diff
@@ -951,6 +963,152 @@ def __get_submissions_in_xml(self, **params):
 
         return (lazy_instance.xml for lazy_instance in queryset)
 
+    def sync_media_files(self):
+        identifier = self.identifier
+        kc_server, kc_path = self.__parse_identifier(identifier)
+        metadata_url = self.external_to_internal_url(
+            '{}/api/v1/metadata'.format(kc_server)
+        )
+
+        def _delete_kc_file(kc_file_: dict, file_: AssetFile = None):
+            """
+            A simple utility to delete file in KC through proxy.
+            If related KPI file is provided (i.e. `file_`), it is deleted too.
+            """
+            # Delete file in KC
+            self._kobocat_request('DELETE',
+                                  url=kc_file_['url'],
+                                  expect_formid=False)
+
+            if file_ is None:
+                return
+
+            # Delete file in KPI if requested
+            file_.delete(force=True)
+
+        def _upload_to_kc(file_):
+            """
+            Prepares request and data corresponding to the kind of media file
+            (i.e. FileStorage or remote URL) to `POST` to KC through proxy.
+            """
+            kwargs = {
+                "data": {
+                    'data_value': file_.metadata['filename'],
+                    'xform': self.xform_id,
+                    'data_type': 'media',
+                    'from_kpi': True,
+                }
+            }
+
+            if file_.is_remote_url:
+                # KC stores url in `data_value`
+                data = {
+                    'data_value': file_.metadata['redirect_url'],
+                    'data_file_type': file_.metadata['mimetype'],
+                    'file_hash': file_.metadata['hash']
+                }
+                kwargs['data'].update(data)
+            else:
+                kwargs['files'] = {
+                    'data_file': (
+                        file_.metadata['filename'],
+                        file_.content.file.read(),
+                        file_.metadata['mimetype'],
+                    )
+                }
+
+            self._kobocat_request('POST',
+                                  url=metadata_url,
+                                  _kobocat_request=False,
```

Comment:
should be `expect_formid`?

Reply:
it is! I think you haven't refreshed your browser 😇

Reply:
✅